### PR TITLE
Feature/tao 3949 async export

### PR DESCRIPTION
**Pull request by antoinerobin**
I'm not sure about the version number.
When you export items / tests with taskqueue extension aren't loaded so constants aren't accessible.
I put them in class constants to avoid this issue, but if you get this version of tao with older version of taoQtiItem / taoItem ... it will break.

### CODE REVIEW

**Comment by antoinerobin** [comment]
@ssipasseuth We just need your review on the javascript part, since you did the taskqueue table

File: actions/class.TaskQueueData.php [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+        /**
+         * @var $taskQueue oat\Taskqueue\Persistence\RdsQueue
+         */
+        $taskQueue = $this->getServiceManager()->get(Queue::SERVICE_ID);
+
+        if(is_a($taskQueue , 'oat\Taskqueue\Persistence\RdsQueue', true)) {
+            /** @var oat\Taskqueue\Action\TaskQueueSearch $dataPayLoad */
+            $dataPayLoad =  $taskQueue->getPayload($user->getIdentifier());
+
+            echo json_encode($dataPayLoad);
```
Comment (jbout) [inline actions/class.TaskQueueData.php]: please use return "$this->returnJson($dataPayLoad)"

File: actions/class.TaskQueueData.php [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    public function getTasks()
+    {
+        $user = common_session_SessionManager::getSession()->getUser();
+
+        /**
+         * @var $taskQueue oat\Taskqueue\Persistence\RdsQueue
+         */
+        $taskQueue = $this->getServiceManager()->get(Queue::SERVICE_ID);
+
+        if(is_a($taskQueue , 'oat\Taskqueue\Persistence\RdsQueue', true)) {
```
Comment (jbout) [inline actions/class.TaskQueueData.php]: Should not check for a specific class name, but a feature defined by an interface

File: actions/class.TaskQueueData.php [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                $file = $directory->getFile($filename);
+                $this->output($file);
+                return;
+            }
+        }
+        return $this->returnJson([
+            'success' => false,
+        ]);
+    }
+    protected function output(File $file)
```
Comment (jbout) [inline actions/class.TaskQueueData.php]: please use tao_helpers_Http::returnStream() instead, to not create temporary files

File: models/classes/TaskQueueActionTrait.php [hunk 4]
```diff
@@ -22,6 +22,8 @@
 
 use oat\oatbox\task\Queue;
 use oat\oatbox\task\Task;
+use oat\oatbox\user\User;
```
Comment (jbout) [inline models/classes/TaskQueueActionTrait.php]: User unused

**Comment by antoinerobin** [comment]
requires : https://github.com/oat-sa/generis/pull/326

**Pull request closed without merging**